ksqldb-functional-tests/src/test/resources/query-validation-tests/greatest.json

```diff
@@ -0,0 +1,178 @@
+{
+  "comments": [
+    "Tests covering the greatest function."
+  ],
+  "tests": [
+    {
+      "name": "test greatest with numeric types",
+      "statements": [
+        "CREATE STREAM INPUT1 (ID BIGINT KEY, I1 INT, I2 INT, I3 INT) WITH (kafka_topic='input_topic_1', value_format='json');",
+        "CREATE STREAM INPUT2 (ID BIGINT KEY, B1 BIGINT, B2 BIGINT, B3 BIGINT) WITH (kafka_topic='input_topic_2', value_format='json');",
+        "CREATE STREAM INPUT3 (ID BIGINT KEY, DE1 DECIMAL(9,3), DE2 DECIMAL(9,3), DE3 DECIMAL(9,3)) WITH (kafka_topic='input_topic_3', value_format='json');",
+        "CREATE STREAM INPUT4 (ID BIGINT KEY, DO1 DOUBLE, DO2 DOUBLE, DO3 DOUBLE) WITH (kafka_topic='input_topic_4', value_format='json');",
+
+        "CREATE STREAM OUTPUT1 AS SELECT ID, GREATEST(I1, I2, I3, null, 5) AS HIGHEST FROM INPUT1;",
+        "CREATE STREAM OUTPUT2 AS SELECT ID, GREATEST(B1, B2, B3, null) AS HIGHEST FROM INPUT2;",
+        "CREATE STREAM OUTPUT3 AS SELECT ID, GREATEST(DE1, DE2, DE3, null) AS HIGHEST FROM INPUT3;",
+        "CREATE STREAM OUTPUT4 AS SELECT ID, GREATEST(DO1, DO2, DO3, null) AS HIGHEST FROM INPUT4;"
+      ],
+      "inputs": [
+        {"topic": "input_topic_1", "key": 1, "value": {"I1": 2147483647, "I¬2": -2147483648, "I3":  null}},
+        {"topic": "input_topic_2", "key": 1, "value": {"B1": 9223372036854775807, "B2": -9223372036854775808, "B3":  null}},
+        {"topic": "input_topic_3", "key": 1, "value": {"DE1": 123456.789, "DE2": -987654.321, "DE3":  null}},
+        {"topic": "input_topic_4", "key": 1, "value": {"DO1": 50000.555, "DO2": -99999.555, "DO3":  null}}
+      ],
+      "outputs": [
+        {"topic": "OUTPUT1", "key": 1, "value": {"HIGHEST": 2147483647}},
+        {"topic": "OUTPUT2", "key": 1, "value": {"HIGHEST": 9223372036854775807}},
+        {"topic": "OUTPUT3", "key": 1, "value": {"HIGHEST": 123456.789}},
+        {"topic": "OUTPUT4", "key": 1, "value": {"HIGHEST": 50000.555}}
+      ],
+      "post": {
+        "sources": [
+          {"name": "OUTPUT1", "type": "stream", "schema": "ID BIGINT KEY, HIGHEST INT"},
+          {"name": "OUTPUT2", "type": "stream", "schema": "ID BIGINT KEY, HIGHEST BIGINT"},
+          {"name": "OUTPUT3", "type": "stream", "schema": "ID BIGINT KEY, HIGHEST DECIMAL(9,3)"},
+          {"name": "OUTPUT4", "type": "stream", "schema": "ID BIGINT KEY, HIGHEST DOUBLE"}
+        ]
+      }
+    },
+    {
+      "name": "test implicit casting to long",
+      "statements": [
+        "CREATE STREAM INPUT (ID BIGINT KEY, N1 INT, N2 BIGINT) WITH (kafka_topic='input_topic',value_format='json');",
+        "CREATE STREAM OUTPUT AS SELECT ID, GREATEST(N1, N2) AS HIGHEST FROM INPUT;"
+      ],
+      "expectedException": {
+        "type": "io.confluent.ksql.util.KsqlStatementException",
+        "message": "Function 'greatest' does not accept parameters (INTEGER, BIGINT)."
```

Comment:
I thought we were expecting a more specific error message about ambiguous casts? (Same question for the other expected errors below.)

Reply:
Yeah, since the error message was changed as part of the UdfIndex changes I forgot to re-test before pushing here as I didn't merge any code. These tests fail now, so rewrote them.